torch_ttnn/passes/lowering/to_tt_pass.py

```diff
@@ -1131,12 +1133,97 @@ def batch_norm_inference(input, weight, bias, mean, var, momentum, eps):
     return gm
 
 
+def DigestAtenOps(gm: torch.fx.GraphModule) -> torch.fx.GraphModule:
+    nodes = list(gm.graph.nodes)
+    for node in nodes:
+        g = GraphWrapper(node)
+
+        def rewrite_node(node):
+            args = node.args
+            kwargs = node.kwargs
+
+            if node.target == torch.ops.aten.index.Tensor:
+
+                def edit_meta_val(node, shape, dtype):
+                    fake_mode = FakeTensorMode()
+                    fake_tensor = fake_mode.from_tensor(torch.zeros(shape, dtype=dtype))
+                    node.meta["val"] = fake_tensor
+
+                def broadcast_indices(indices):
+                    import numpy as np
+
+                    indices_shapes = [get_shape(indices[i]) for i in range(len(indices))]
+                    broadcasted_shape = torch.Size(np.broadcast_shapes(*indices_shapes))
+                    broadcasted_indices = []
+                    for i in range(len(indices)):
+                        if indices_shapes[i] == broadcasted_shape:
+                            broadcasted_indices.append(indices[i])
+                        else:
+                            broadcasted_indices.append(
+                                g.call_function(torch.ops.aten.expand.default, (indices[i], broadcasted_shape))
+                            )
+                            edit_meta_val(broadcasted_indices[-1], broadcasted_shape, indices[i].meta["val"].dtype)
```

Comment:
Maybe we can add a new `call_function_with_new_shape` to `GraphWrapper` and set the new shape in the meta? So we don't need to update the meta separately

I'm thinking something like `def call_function_with_new_shape(self, target, new_shape, args, kwargs)`